apps/avifenc.c

```diff
@@ -574,8 +574,10 @@ static avifBool avifImageSplitGrid(const avifImage * gridSplitImage, uint32_t gr
     uint32_t cellWidth, cellHeight;
     avifPixelFormatInfo formatInfo;
     avifGetPixelFormatInfo(gridSplitImage->yuvFormat, &formatInfo);
-    if (!avifGetBestCellSize("horizontally", gridSplitImage->width, gridCols, formatInfo.chromaShiftX, &cellWidth) ||
-        !avifGetBestCellSize("vertically", gridSplitImage->height, gridRows, formatInfo.chromaShiftY, &cellHeight)) {
+    const avifBool isSubsampledX = !formatInfo.monochrome && formatInfo.chromaShiftX;
+    const avifBool isSubsampledY = !formatInfo.monochrome && formatInfo.chromaShiftY;
+    if (!avifGetBestCellSize("horizontally", gridSplitImage->width, gridCols, isSubsampledX, &cellWidth) ||
+        !avifGetBestCellSize("vertically", gridSplitImage->height, gridRows, isSubsampledY, &cellHeight)) {
```

Comment:
I spotted another bug fix (actually two) in tests/gtest/avifincrtest_helpers.cc. See my comment there.

Since this pull request has a lot of cleanup changes, the commit message should highlight the bug fixes. It may even make sense to move the bug fixes to a separate pull request.